integration-test/common/conftest.py

```diff
@@ -0,0 +1,76 @@
+import pytest
+import os
+import tempfile
+import shutil
+import subprocess
+
+RUN_SCRIPT_FNAME = "run.sh"
+OUTPUT_DIR = "output_dir"
+SWEEP_GENERATOR = "generate_design_sweeps.py"
+
+def is_master(config):
+  """Return true we're on a master node."""
+  return not hasattr(config, 'workerinput')
+
+def pytest_addoption(parser):
+  parser.addoption("--sweep-file", action="store", help="Sweep file")
+
+def pytest_configure(config):
+  """Generate sweeps using the sweep generator.
+
+  Only the master node performs the sweep generation.
+  """
+  if is_master(config):
+    cwd = os.getcwd()
+    config.tmp_output_dir = tempfile.mkdtemp(dir=cwd)
+    _, config.tmp_sweep_file = tempfile.mkstemp(suffix=".xe", dir=cwd)
+    # Read the sweep file and update the `output_dir` field with a temp dir.
+    sweep_file = config.getoption("--sweep-file")
+    if not os.path.isabs(sweep_file):
+      sweep_file = os.path.join(cwd, sweep_file)
+    with open(sweep_file, "r") as f_in, open(config.tmp_sweep_file,
+                                             "w") as f_out:
+      fmt_key = "%%(%s)s" % OUTPUT_DIR
```

Comment:
Wouldn't it be easier to just read the entire file as a format string? I just looked through the config file and the only place a "%" appears is in the `set output_dir` command. We can just use that as a template and write a new sweep file into the desired location. Then we don't have to go through this line by line (which is really doing the same thing).

Reply:
That's good advice. Done as suggested.